components/director/internal/domain/application/repository_test.go

```diff
@@ -15,39 +20,319 @@ import (
+			InputPageSize:  2,
+			InputCursor:    "",
+			ExpectedOffset: 0,
+			ExpectedLimit:  limit,
+			Rows: sqlmock.NewRows([]string{"id", "tenant_id", "name", "description", "status_condition", "status_timestamp", "healthcheck_url"}).
+				AddRow(app1ID, tenantID, "App ABC", "Description for application ABC", "INITIAL", timestamp, "http://domain.local/app1").
+				AddRow(app2ID, tenantID, "App XYZ", "Description for application XYZ", "INITIAL", timestamp, "http://domain.local/app2"),
+			TotalCount: 2,
+		},
+		{
+			Name:           "Success getting next page",
+			InputPageSize:  2,
+			InputCursor:    pagination.EncodeNextOffsetCursor(1, 2),
+			ExpectedOffset: offset,
+			ExpectedLimit:  limit,
+			Rows: sqlmock.NewRows([]string{"id", "tenant_id", "name", "description", "status_condition", "status_timestamp", "healthcheck_url"}).
+				AddRow(app1ID, tenantID, "App ABC", "Description for application ABC", "INITIAL", timestamp, "http://domain.local/app1").
+				AddRow(app2ID, tenantID, "App XYZ", "Description for application XYZ", "INITIAL", timestamp, "http://domain.local/app2"),
+			TotalCount: 2,
+		}}
+	for _, testCase := range testCases {
+		t.Run(testCase.Name, func(t *testing.T) {
+			//GIVEN
+			sqlxDB, sqlMock := testdb.MockDatabase(t)
+			defer sqlMock.AssertExpectations(t)
+			ctx := persistence.SaveToContext(context.TODO(), sqlxDB)
+
+			pgRepository := application.NewPostgresRepository(conv)
+
+			expectedQuery := fmt.Sprintf(pageableQuery, testCase.ExpectedLimit, testCase.ExpectedOffset)
+			sqlMock.ExpectQuery(expectedQuery).
+				WithArgs(tenantID).
+				WillReturnRows(testCase.Rows)
+
+			countRow := sqlMock.NewRows([]string{"count"}).AddRow(testCase.TotalCount)
+			sqlMock.ExpectQuery(countQuery).
+				WithArgs(tenantID).
+				WillReturnRows(countRow)
+
+			//THEN
+			modelPage, err := pgRepository.List(ctx, tenantID, nil, testCase.InputPageSize, testCase.InputCursor)
+
+			//THEN
+			require.NoError(t, err)
+			assert.Equal(t, testCase.ExpectedLimit, modelPage.TotalCount)
+			require.NoError(t, sqlMock.ExpectationsWereMet())
+
+			assert.Equal(t, app1ID, modelPage.Data[0].ID)
+			assert.Equal(t, tenantID, modelPage.Data[0].Tenant)
+
+			assert.Equal(t, app2ID, modelPage.Data[1].ID)
+			assert.Equal(t, tenantID, modelPage.Data[1].Tenant)
+
+		})
+	}
+
+	t.Run("Returns error when decoded cursor is non-positive number", func(t *testing.T) {
+		//GIVEN
+		sqlxDB, sqlMock := testdb.MockDatabase(t)
+		defer sqlMock.AssertExpectations(t)
+		ctx := persistence.SaveToContext(context.TODO(), sqlxDB)
+		pgRepository := application.NewPostgresRepository(conv)
+		//THEN
+		_, err := pgRepository.List(ctx, tenantID, nil, 2, pagination.EncodeNextOffsetCursor(1, -3))
+
+		//THEN
+		require.EqualError(t, err, "while decoding page cursor: cursor is not correct")
+	})
+}
+
+func TestPgRepository_GetByID_ShouldReturnApplicationModelForApplicationEntity(t *testing.T) {
+	// given
+	conv := application.NewConverter(nil, nil, nil, nil)
```

Comment:
Test the following cases:
- Success
- Converter Error
- DB Error

Please see document / fetch request repository tests.